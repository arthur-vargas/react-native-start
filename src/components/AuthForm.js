import React, { useState } from 'react';
import {
  StyleSheet,
  Text,
  Button,
  TextInput,
} from 'react-native';
import PropTypes from 'prop-types';
import Spacer from './Spacer';

const AuthForm = ({
  headerText,
  errorMessage,
  onSubmit,
  submitButtonText,
}) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  return (
    <>
      <Spacer>
        <Text h3>{headerText}</Text>
      </Spacer>
      <Text>Login</Text>
      <TextInput
        value={username}
        onChangeText={setUsername}
        autoCapitalize="none"
        autoCorrect={false}
      />
      <Spacer />
      <Text>Password</Text>
      <TextInput
        secureTextEntry
        value={password}
        onChangeText={setPassword}
        autoCapitalize="none"
        autoCorrect={false}
      />
      {errorMessage ? (
        <Text style={styles.errorMessage}>{errorMessage}</Text>
      ) : null}
      <Spacer>
        <Button
          title={submitButtonText}
          onPress={() => onSubmit({ username, password })}
        />
      </Spacer>
    </>
  );
};

const styles = StyleSheet.create({
  errorMessage: {
    fontSize: 16,
    color: 'red',
    marginLeft: 15,
    marginTop: 15,
  },
});

AuthForm.propTypes = {
  headerText: PropTypes.string.isRequired,
  errorMessage: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  submitButtonText: PropTypes.string.isRequired,
};

export default AuthForm;
