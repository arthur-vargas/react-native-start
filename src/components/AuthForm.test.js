/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {
  Button,
} from 'react-native';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import AuthForm from './AuthForm';

describe('Component: AuthForm', () => {
  const props = {
    headerText: '',
    errorMessage: '',
    onSubmit: jest.fn(async () => {
      // console.log('call signin');
    }),
    submitButtonText: '',
  };

  beforeEach(() => {
    props.onSubmit.mockClear();
  });

  it('Render', () => {
    expect(
      shallow(<AuthForm {...props} />),
    ).toMatchSnapshot();
  });

  it('Don\'t onSubmit', () => {
    shallow(<AuthForm {...props} />);
    expect(props.onSubmit).toHaveBeenCalledTimes(0);
  });

  it('Call onSubmit', async () => {
    const wrapper = shallow(<AuthForm {...props} />);
    await wrapper.find(Button).first().simulate('press');
    expect(props.onSubmit).toHaveBeenCalledTimes(1);
  });
});
