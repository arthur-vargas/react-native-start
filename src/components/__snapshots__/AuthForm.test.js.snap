// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AuthForm Render 1`] = `ShallowWrapper {}`;
